### PR TITLE
Feature Date input

### PR DESCRIPTION
**Pull request by chromale**

### CODE REVIEW

File: src/DateInput/helpers/validate.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const daysInMonth = (m: any, y: any) => {
+  switch (m) {
+    case 1:
```
Comment (tomzaoral) [inline src/DateInput/helpers/validate.tsx]: 1 is February and 0 is January? Or this is wrong and should be 2?

File: src/DateInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  }, [value])
+
+  const labels = inputLabels || {
+    day: 'Day',
+    month: 'Month',
+    year: 'Year'
+  }
+
+  const DEFAULT_MONTHS = [
+    { value: 1, label: 'January' },
```
Comment (tomzaoral) [inline src/DateInput/index.tsx]: This does not add up with date validation I mentioned above. Please double check it.

File: src/DateInput/index.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        <Select
+          name="month"
+          label={labels.month}
+          value={value.month}
+          onChange={(option: any) => setValue({ ...value, month: option })}
+          options={months || DEFAULT_MONTHS}
+        />
+        <Input
+          name="year"
+          min="1"
```
Comment (tomzaoral) [inline src/DateInput/index.tsx]: I suggest to set minimum to (current_year - 18 - 100) and maximum to (current_year - 18).

For example:
minimum = 1901
maximum = 2001 (excluding underaged ppl)

File: src/DateInput/helpers/validate.tsx (hunk 1 above)
Comment (chromale) [reply src/DateInput/helpers/validate.tsx]: It works fine. I’ve double-tested it. `isValidDate()` has its own logic which is not dependents on default months.

File: src/DateInput/index.tsx (hunk 3 above)
Comment (chromale) [reply src/DateInput/index.tsx]: I believe we can do this validation directly in FE app and keep this component pure for now. It’s just handling the correct date, additional logic and more functions can be done later when there is more time :)

**Pull request merged**